tests/components/humidifier/test_intent.py

```diff
@@ -0,0 +1,208 @@
+"""Tests for the humidifier intents."""
+from homeassistant.components.humidifier import (
+    ATTR_AVAILABLE_MODES,
+    ATTR_HUMIDITY,
+    ATTR_MODE,
+    DOMAIN,
+    SERVICE_SET_HUMIDITY,
+    SERVICE_SET_MODE,
+    intent,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_SUPPORTED_FEATURES,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.helpers.intent import IntentHandleError
+
+from tests.common import async_mock_service
+
+
+async def test_intent_set_humidity(hass):
+    """Test the set humidity intent."""
+    hass.states.async_set(
+        "humidifier.bedroom_humidifier", STATE_ON, {ATTR_HUMIDITY: 40}
+    )
+    humidity_calls = async_mock_service(hass, DOMAIN, SERVICE_SET_HUMIDITY)
+    turn_on_calls = async_mock_service(hass, DOMAIN, SERVICE_TURN_ON)
+    await intent.async_setup_intents(hass)
+
+    result = await hass.helpers.intent.async_handle(
+        "test",
+        intent.INTENT_HUMIDITY,
+        {"name": {"value": "Bedroom humidifier"}, "humidity": {"value": "50"}},
+    )
+    await hass.async_block_till_done()
+
+    assert result.speech["plain"]["speech"] == "The bedroom humidifier is set to 50%"
+
+    assert len(turn_on_calls) == 0
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)